ui/uihandler.go

```diff
@@ -922,7 +916,7 @@ func Report(baseurl, port, staticRsrcDir string, datadir string) {
 	}
 	fsd := http.FileServer(http.Dir(dataDir))
 	http.Handle(uiPath+"data/", LogAndFilterDataRequest(http.StripPrefix(uiPath+"data", fsd)))
-	if err := http.ListenAndServe(nPort, nil); err != nil {
+	if err := http.ListenAndServe(hostPort, nil); err != nil {
```

Comment:
this is a bug I think, ie it will listen only on localhost instead of all addr

Reply:
I am not seeing the bug: https://gist.github.com/danehans/334fca847547bf9827bd5e87f1c768c2#file-sethostandport_testing-md

Reply:
try using the external ip - 0.0.0.0 as a client is just an alias for localhost
what the default ":8080" means as a server socket is to bind on all addresses

for me it would be to try to curl 100.123.127.245:8080, use ifconfig -a to get your machine's non localhost ip

Reply:
you can also do "lsof -p pidoffortio" to see what it's bound on

Reply:
After going through the code again, I think i see the bug you're talking about.

Reply:
This is what I see with the updated PR that I am going to push:

Test 1: gcloud VM eth0 IP 10.138.0.2:
```
$ sudo docker run -d --net host danehans/fortio:listen_addr server -http-port 10.138.0.2:8080

$ sudo docker logs 60df1944d635
Fortio 0.6.8 grpc ping server listening on port 8079
UI starting - visit:
http://10.138.0.2:8080/fortio/
23:53:34 I uihandler.go:91> Using resources directory set at link time: /usr/local/lib/fortio
Fortio 0.6.8 echo server listening on port 10.138.0.2:8080
Https redirector running on :8081

$ curl http://10.138.0.2:8080/fortio/
<SNIP>
URL: <input type="text" name="url" size="80" value="http://10.138.0.2:8080/echo?delay=250us:30%,5ms:5%&status=503:0.5%,429:1.5%" /> <br />
<SNIP>

$ netstat -ant | grep 8080
tcp        0      0 10.138.0.2:8080         0.0.0.0:*               LISTEN     

$ curl http://localhost:8080/fortio/
curl: (7) Failed to connect to localhost port 8080: Connection refused

$ sudo lsof -p 14585
COMMAND   PID USER   FD      TYPE DEVICE SIZE/OFF   NODE NAME
fortio  14585 root  cwd       DIR   0,45     4096 775113 /
fortio  14585 root  rtd       DIR   0,45     4096 775113 /
fortio  14585 root  txt       REG   0,45  8074880 775103 /usr/local/bin/fortio
fortio  14585 root  mem       REG    8,1          775103 /usr/local/bin/fortio (stat: No such file or directory)
fortio  14585 root    0r     FIFO   0,12      0t0  46915 pipe
fortio  14585 root    1w     FIFO   0,12      0t0  46916 pipe
fortio  14585 root    2w     FIFO   0,12      0t0  46917 pipe
fortio  14585 root    3u     IPv6  46999      0t0    TCP *:8079 (LISTEN)
fortio  14585 root    4u  a_inode   0,13        0  10027 [eventpoll]
fortio  14585 root    5u     IPv6  47484      0t0    TCP *:tproxy (LISTEN)
fortio  14585 root    6u     IPv4  47007      0t0    TCP fortio-vm.c.level-scheme-173421.internal:http-alt (LISTEN)
```

Test 2: Setting `-http-port 8080` only:
```
$ sudo docker run -d --net host danehans/fortio:listen_addr server -http-port 8080
707b5ac6bdc8e567235f250d208f3f4c0b1f806ce546d7c243a7fdeae098cf76

$ sudo docker logs 707b5ac6bdc8
Fortio 0.6.8 grpc ping server listening on port 8079
UI starting - visit:
http://localhost:8080/fortio/
00:01:09 I uihandler.go:91> Using resources directory set at link time: /usr/local/lib/fortio
Fortio 0.6.8 echo server listening on port :8080
Https redirector running on :8081

$ curl http://localhost:8080/fortio/
<SNIP>
Title/Labels: <input type="text" name="labels" size="40" value="Fortio" /> (empty to skip title)<br />
URL: <input type="text" name="url" size="80" value="http://localhost:8080/echo?delay=250us:30%,5ms:5%&status=503:0.5%,429:1.5%" /> <br />
<SNIP>

$ netstat -ant | grep 8080
tcp        0      0 127.0.0.1:41134         127.0.0.1:8080          TIME_WAIT  
tcp6       0      0 :::8080                 :::*                    LISTEN     

$ curl http://10.138.0.2:8080/fortio/
 <SNIP>
URL: <input type="text" name="url" size="80" value="http://localhost:8080/echo?delay=250us:30%,5ms:5%&status=503:0.5%,429:1.5%" /> <br />
<SNIP>

$ curl http://127.0.0.1:8080/fortio/
<SNIP>
URL: <input type="text" name="url" size="80" value="http://localhost:8080/echo?delay=250us:30%,5ms:5%&status=503:0.5%,429:1.5%" /> <br />
<SNIP>

$ sudo lsof -p 14378
COMMAND   PID USER   FD      TYPE DEVICE SIZE/OFF   NODE NAME
fortio  14378 root  cwd       DIR   0,45     4096 775113 /
fortio  14378 root  rtd       DIR   0,45     4096 775113 /
fortio  14378 root  txt       REG   0,45  8074880 775103 /usr/local/bin/fortio
fortio  14378 root  mem       REG    8,1          775103 /usr/local/bin/fortio (stat: No such file or directory)
fortio  14378 root    0r     FIFO   0,12      0t0  45785 pipe
fortio  14378 root    1w     FIFO   0,12      0t0  45786 pipe
fortio  14378 root    2w     FIFO   0,12      0t0  45787 pipe
fortio  14378 root    3u     IPv6  45845      0t0    TCP *:8079 (LISTEN)
fortio  14378 root    4u  a_inode   0,13        0  10027 [eventpoll]
fortio  14378 root    5u     IPv6  45853      0t0    TCP *:http-alt (LISTEN)
fortio  14378 root    6u     IPv6  45855      0t0    TCP *:tproxy (LISTEN)
```

Reply:
I believe this is the expected behavior, so I'm going to submit my updated PR.

Reply:
that's strange, I must be misreading the code, but it seems it would use "localhost:8080" for Serve which should only bind on localhost ? (like if you used -http-port localhost:8080)